programs/castle-lending-aggregator/src/state.rs

```diff
@@ -55,6 +60,16 @@ pub struct Vault {
     /// Mint address of the tokens that are stored in vault
     pub reserve_token_mint: Pubkey,
 
+    /// Whether to run rebalance as a proof check or a calculation
+    pub rebalance_mode: RebalanceMode,
+
+    /// Strategy type that is executed during rebalance
+    pub strategy_type: StrategyType,
```

Comment:
It may be prudent (pedantic?) to explicitly set the layout/size for enums used in this struct, e.g.

```Rust
#[repr(u8)]
pub enum StrategyType {
    MaxYield,
    EqualAllocation,
}
```
I say pedantic because I recognize that
1. The `assert_size` macro will very likely catch this case
2. As-is the enums will __very likely__ be  `repr(u8)` anyhow (although maybe not guaranteed :thinking: )